### ISSUE TITLE
should(obj).have.only.keys() do not work when using should/as-function

### ISSUE THREAD

**Issue by fulax** [shouldjs/should.js#177]
Hi.

The recently introduced feature `.only` in `keys()` assertion does not work in function mode. I was able to write a minimum test case, given below.

Current output:
```
/home/user/work/ext/shouldjs/node_modules/should/as-function.js:359
        throw e;
        ^

TypeError: Cannot read property 'have' of undefined
    at Assertion.<anonymous> (/home/user/work/ext/shouldjs/node_modules/should/as-function.js:2337:18)
    at Assertion.value [as keys] (/home/user/work/ext/shouldjs/node_modules/should/as-function.js:337:14)
    at Object.<anonymous> (/home/user/work/ext/shouldjs/test.js:4:30)
```

I was able to change the line 2337 of the `as-function.js`  to make it work by replacing

```diff
     if (this.onlyThis) {
-      obj.should.have.size(keys.length);
+      should(obj).have.size(keys.length);
     }
```

## Node.js version?
v8.11.3

## Should.js version?
13.2.2

## How to reproduce it?
```javascript
'use strict';
const should = require('should/as-function');
should({ 'a': 1 }).have.keys('a'); // OK
should({ 'a': 1 }).have.only.keys('a'); // Throws
```

**Comment by btd** [comment]
Yep you are right.

**Comment by btd** [comment]
This is fixed in 13.2.3
